packages/ui-kit/scripts/parse-design-tokens.mts

```diff
@@ -0,0 +1,342 @@
+import fs, { promises as fsPromises } from 'fs'
+import path from 'path'
+import slugify from 'slugify'
+import ora from 'ora'
+
+const GENERATED_WARNING =
+  'This file is generated automatically by scripts/parse-design-tokens.js. Do not edit manually.'
+
+// Parse typography
+const valueKeyPropMapping: Record<string, string> = {
+  fontSize: 'font-size',
+  fontFamily: 'font-family',
+  fontWeight: 'font-weight',
+  lineHeight: 'line-height',
+  letterSpacing: 'letter-spacing'
+}
+
+// Map font weight from string to number
+const weightMapping: Record<string, string> = {
+  Thin: '100',
+  'Extra Light': '200',
+  Light: '300',
+  Regular: '400',
+  Medium: '500',
+  'Semi Bold': '600',
+  Bold: '700',
+  'Extra Bold': '800',
+  Black: '900'
+}
+
+type VariableProps = {
+  name: string
+  type: string
+  isAlias: boolean
+  value: any
+}
+
+type TokenDataProps = {
+  cssName: string
+  jsName: string
+  kind?: 'typography'
+  type: 'number' | 'string'
+  value: string
+}
+
+type ThemeProps = {
+  name: string
+  variables: VariableProps[]
+}
+
+type VariablesJSONProps = {
+  collections: {
+    name: string
+    modes: ThemeProps[]
+  }[]
+}
+
+// Slugify string with strict mode
+const slugifyStr = (str: string): string => slugify(str.replaceAll('/', '-'), { lower: true, strict: true })
+
+// Convert kebab-case to camelCase
+const kebabCaseToCamelCase = (kebabStr: string): string =>
+  kebabStr
+    .split('-')
+    .map((word, index) => (index === 0 ? word : word[0].toUpperCase() + word.slice(1)))
+    .join('')
+
+// Convert camelCase to kebab-case
+const camelCaseToKebabCase = (camelStr: string): string => camelStr.replace(/([a-z0-9])([A-Z])/g, '$1-$2').toLowerCase()
+
+// Parse Figma variable value
+const parseValue = (variable: VariableProps): string => {
+  const { type, value } = variable
+  switch (type) {
+    case 'color':
+      return `rgba(${value.r}, ${value.g}, ${value.b}, ${value.a})`
+    case 'number':
+      return `${value}px`
+    case 'effect': {
+      const { offset, radius, spread, color } = value.effects[0]
```

Comment:
Should we iterate all the effects? Why using only the first one?

Reply:
This is a basic shadow. We can change it later if we get a more advanced case.